### DISCUSSION TITLE
[lodash] maxBy shouldn't return undefined for non-empty arrays

### DISCUSSION THREAD

**Discussion by NimaiMalle** [DefinitelyTyped/DefinitelyTyped#68124] [Issues with a @types package] [answered]
Feels like there should be an explicit overload for non-empty arrays, like so:

```typescript
interface LoDashStatic {
    // Overloaded signature for a non-empty array (undefined isn't explicitly included)
    maxBy<T>(collection: NonEmptyArray<T>, iteratee?: ValueIteratee<T>): T;

    // Existing signature for array that may be empty
    maxBy<T>(collection: List<T> | null | undefined, iteratee?: ValueIteratee<T>): T | undefined;
}
```

Here, I'm assuming a non-empty array type something like this, or `[T,...T[]]` could be in-lined above.
```typescript
// Define a type that represents a non-empty array
type NonEmptyArray<T> = [T, ...T[]];
```

[a comment by an automated account is not shown]

**Comment by aj-r** [comment, marked as answer]
~We actually tried this, but unfortunately the tuple type causes some unexpected issues in other use cases, so we had to revert it.~

Edit: this is wrong, see below.

**Reply by NimaiMalle** [reply]
Ah, bummer!  Do you happen to still have what was tried, and what test cases failed?  Is there an old thread or something?  I'm curious!  Thank you.

**Reply by aj-r** [reply]
Actually it looks like I'm wrong. We did something like this in #66459, and there seems to be no issues. I must have mixed this up with something else.

Feel free to send a PR.
